lib/jxl/encode.cc

```diff
@@ -490,27 +507,28 @@ JxlEncoderStatus JxlEncoderSetBasicInfo(JxlEncoder* enc,
         "other exponent bits per sample combinations than IEEE binary32 and "
         "binary16 not (yet) supported");
```

Comment:
They are actually supported. In VarDCT mode, this info is just metadata. In Modular mode, it is supported.

What is not supported is setting/getting images to/from the encoder/decoder with buffers that are not binary16, binary32, uint8 or uint16. But that's unrelated: that's a limitation of JxlPixelFormat. But there's nothing stopping you from using binary32 buffers to work with e.g. fp24 images (7 exp bits, 16 mantissa bits, 1 sign bit).

So I would suggest to simplify this whole part and just set the metadata bits_per_sample and exponent_bits_per_sample directly - if the new check above didn't return error, then the values should be fine.

Reply:
Not using the `Set***Samples` functions anymore, but setting everything directly. Functionality should be unchanged,